### PR TITLE
additional use of Java streaming API

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/edu/tamu/app/service/manager/GitHubProjectService.java [hunk 1]
```diff
@@ -26,46 +24,43 @@ public GitHubProjectService(final ManagementService managementService) throws IO
...
+            toProductName(project),
+            productName,
+            getCards(project)
+        );
+    }
+
     private List<Card> getCards(GHProject project) {
         return exceptionHandlerWrapper(project, i -> i.listColumns().asList().stream())
             .flatMap(c -> exceptionHandlerWrapper(c, i -> i.listCards().asList().stream()))
-            .collect(HashMap<GHProjectCard, GHIssue>::new, (m, c) -> m.put(c, exceptionHandlerWrapper(c, i -> i.getContent())), HashMap<GHProjectCard, GHIssue>::putAll)
```
Comment (ghost) [inline src/main/java/edu/tamu/app/service/manager/GitHubProjectService.java:54]: No reason to collect and then stream again.

File: src/main/java/edu/tamu/app/service/manager/AbstractGitHubService.java [hunk 2]
```diff
@@ -183,23 +179,13 @@ protected boolean hasLabelByName(final Collection<GHLabel> labels, final String
 
     protected String getCardType(final GHIssue content) throws IOException {
         final List<GHLabel> labels = (List<GHLabel>) content.getLabels();
-        final GHLabel label = labels.stream()
```
Comment (ghost) [inline src/main/java/edu/tamu/app/service/manager/AbstractGitHubService.java:186]: Not necessary to stream labels 4 times.

File: src/main/java/edu/tamu/app/service/manager/AbstractGitHubService.java [hunk 3]
```diff
@@ -56,8 +60,6 @@
 
     protected final RestTemplate restTemplate;
 
-    protected GHLabel label;
```
Comment (ghost) [inline src/main/java/edu/tamu/app/service/manager/AbstractGitHubService.java:59]: Should not have transient class properties on singletons that are set and used in various arbitrary functions.

File: src/main/java/edu/tamu/app/service/manager/GitHubProjectService.java (hunk 1 above)
Comment (ghost) [reply src/main/java/edu/tamu/app/service/manager/GitHubProjectService.java:54]: This collect to a map with key GHProjectCard was causing loss of a sprint card due to key overwrite of a duplicate GHProjectCard.

```
-private static final GHProjectCard TEST_CARD1 = mock(GHProjectCard.class);
+private static final GHProjectCard TEST_CARD1 = mock(GHProjectCard.class, RETURNS_DEEP_STUBS.get());
```

**Review by kaladay — APPROVED** [review]

**Review by rladdusaw — APPROVED** [review]

**Pull request merged**